sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientConfiguration.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.ConsistencyLevel
+import com.azure.cosmos.BridgeInternal
+
+private[spark] case class CosmosClientConfiguration (
+    endpoint: String,
+    key: String,
+    consistencyLevel: ConsistencyLevel)
+
+private[spark] object CosmosClientConfiguration extends CosmosLoggingTrait {
+    private[spark] def apply(config: Map[String, String]): CosmosClientConfiguration = {
+        val cosmosAccountConfig = CosmosAccountConfig.parseCosmosAccountConfig(config)
+        val consistency = cosmosAccountConfig.consistency match {
+            case Some(consistencyAsString) =>
+                parseStringAsConsistencyLevel(consistencyAsString) match {
+                    case Some(parsedConsistency) => parsedConsistency
+                    // Should we throw instead of defaulting to Eventual if the value does not map correctly?
+                    case None => ConsistencyLevel.EVENTUAL
+                }
+            case None => ConsistencyLevel.EVENTUAL
+        }
+        CosmosClientConfiguration(
+            cosmosAccountConfig.endpoint,
+            cosmosAccountConfig.key,
+            consistency)
+    }
+
+    private def parseStringAsConsistencyLevel(stringValue: String): Option[ConsistencyLevel] = {
+        try {
+            val result = BridgeInternal.fromServiceSerializedFormat(stringValue)
+            Option(result)
+        }
+        catch {
+            case _: IllegalArgumentException =>
+                logWarning(s"Could not parse Consistency value $stringValue in configuration")
+                // ignore the exception and return the default
+                None
```

Comment:
on the parsing failure case we should throw error to avoid unintentional spelling errors in the configuration by the customer

see this as example:
https://github.com/Azure/azure-sdk-for-java/blob/e245776173483d1f90383b821f2f5b123eb283de/sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala#L81-L91

Reply:
Not relevant, code does not exist anymore